lib/ladybug/energy_model/ideal_air_system.rb

```diff
@@ -0,0 +1,99 @@
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'ladybug/energy_model/extension'
+require 'ladybug/energy_model/model_object'
+
+module Ladybug
+  module EnergyModel
+    class IdealAirSystem < ModelObject
+      attr_reader :errors, :warnings
+  
+      def initialize(hash = {})
+        super(hash)
+        raise "Incorrect model type '#{@type}'" unless @type == 'IdealAirSystem'
+      end
+    
+      def defaults
+        result = {}
+        result
+      end
+
+      def create_openstudio_object(openstudio_model)       
+        openstudio_ideal_air = OpenStudio::Model::ZoneHVACIdealLoadsAirSystem.new(openstudio_model)
+        if @hash[:heating_limit]
+          openstudio_ideal_air.setHeatingLimit('LimitCapacity')
+          if @hash[:heating_limit] == 'autocalculate'
+            openstudio_ideal_air.autosizeMaximumSensibleHeatingCapacity()
+          elsif 
+            openstudio_ideal_air.setMaximumSensibleHeatingCapacity(@hash[:heating_limit])
+          end
+        end
+        if @hash[:cooling_limit]
+          openstudio_ideal_air.setCoolingLimit('LimitFlowRateAndCapacity')
+          if @hash[:cooling_limit] == 'autocalculate'
+            openstudio_ideal_air.autosizeMaximumTotalCoolingCapacity ()
+            openstudio_ideal_air.autosizeMaximumCoolingAirFlowRate()
+          elsif
+            openstudio_ideal_air.setMaximumTotalCoolingCapacity(@hash[:cooling_limit])
+            openstudio_ideal_air.autosizeMaximumCoolingAirFlowRate()
+          end
+        end
+        if @hash[:economizer_type]
+          openstudio_ideal_air.setOutdoorAirEconomizerType(@hash[:economizer_type])
+        else
+          openstudio_ideal_air.setOutdoorAirEconomizerType(@@schema[:definitions][:IdealAirSystem][:properties][:economizer_type][:default])
+        end
+        if @hash[:sensible_heat_recovery]
+          openstudio_ideal_air.setSensibleHeatRecoveryEffectiveness(@hash[:sensible_heat_recovery])
+        else
+          openstudio_ideal_air.setSensibleHeatRecoveryEffectiveness(@@schema[:definitions][:IdealAirSystem][:properties][:sensible_heat_recovery][:default]) #TODO : Openstudio defaults are different from schema
+        end
+        if @hash[:latent_heat_recovery]
+          openstudio_ideal_air.setLatentHeatRecoveryEffectiveness(@hash[:latent_heat_recovery])
+        else
+          openstudio_ideal_air.setLatentHeatRecoveryEffectiveness(@@schema[:definitions][:IdealAirSystem][:properties][:latent_heat_recovery][:default]) #TODO : Openstudio defaults are different from schema
+        end
+        if @hash[:demand_control_ventilation]
+          if @hash[:demand_control_ventilation] == true
+            openstudio_ideal_air.setDemandControlledVentilationType('OccupancySchedule') #TODO: when true, what is demand control vent. type?
```

Comment:
@chriswmackey Is the demand_control_ventilation type `OccupancySchedule` when true?

Reply:
Yes. Here is the Python code:
https://github.com/ladybug-tools/honeybee-energy/blob/master/honeybee_energy/idealair.py#L286